packages/repository/src/__tests__/fixtures/mixins/findByTitleRepoMixin.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Constructor} from '@loopback/context';
+import {CrudRepository, Model, Where} from '../../../index';
+import {FindByTitleInterface} from './findByTitleInterface';
+
+/*
+ * This function adds a new method 'findByTitle' to a repository class
+ * where 'M' is a model and 'R' is DefaultCrudRepository
+ */
+export function FindByTitleRepositoryMixin<
+  M extends Model & {title: string},
+  R extends Constructor<CrudRepository<M>>
+>(superClass: R) {
```

Comment:
It can be simplified as:
```ts
export function FindByTitleRepositoryMixin<M extends Model & {title: string}>(
  superClass: Constructor<CrudRepository<M>>,
) {
```

Reply:
> It can be simplified as:
> 
> ```ts
> export function FindByTitleRepositoryMixin<M extends Model & {title: string}>(
>   superClass: Constructor<CrudRepository<M>>,
> ) {
> ```

It turns out your new proposal works at runtime, but the compiler is not aware of certain methods like repo.exists or repo.findById. 

This was discovered when @bajtos suggested tests for these methods too.

Then we re-worked the repo mixin to your original suggestion last week:

```
export function FindByTitleRepositoryMixin<
  E extends Entity & {title: string},
  ID,
  Relations extends object = {}
>(superClass: Constructor<EntityCrudRepository<E, ID, Relations>>) {
  return class extends superClass implements FindByTitle<E> {
    async findByTitle(title: string): Promise<E[]> {
      const where = {title} as Where<E>;
      const titleFilter = {where};
      return this.find(titleFilter);
    }
  };
}
```